exporter/datadogexporter/translate_traces.go

```diff
@@ -465,14 +465,48 @@ func getSpanErrorAndSetTags(s pdata.Span, tags map[string]string) int32 {
 	}
 
 	if isError == errorCode {
-		tags[ext.ErrorType] = "ERR_CODE_" + strconv.FormatInt(int64(status.Code()), 10)
+		// Finds the last exception event in the span, and surfaces it to DataDog. DataDog spans only support a single
+		// exception per span, but otel supports many exceptions as "Events" on a given span. The last exception was
+		// chosen for now as most otel-instrumented libraries (http, pg, etc.) only capture a single exception (if any)
+		// per span. If multiple exceptions are logged, it's my assumption that the last exception is most likely the
+		// exception that escaped the scope of the span.
+		//
+		// TODO:
+		//  Seems that the spec has an attribute that hasn't made it to the collector yet -- "exception.escaped".
+		//  This seems optional (SHOULD vs. MUST be set), but it's likely that we want to bubble up the exception
+		//  that escaped the scope of the span ("exception.escaped" == true) instead of the last exception event
+		//  in the case that these events differ.
+		//
+		//  https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/exceptions.md#attributes
+		attribs := pdata.NewAttributeMap()
```

Comment:
I believe `nil` would work ok even if an error isn't found, to avoid this unused allocation

Reply:
Not sure I understand this suggestion -- AttributeMap zero-value does not work, nor does a nil pointer as it's child member (it dereferences in the `Get` call).

Snippet from the pdata package:

```go
// NewAttributeMap creates a AttributeMap with 0 elements.
func NewAttributeMap() AttributeMap {
	orig := []otlpcommon.KeyValue(nil)
	return AttributeMap{&orig}
}

func (am AttributeMap) Get(key string) (AttributeValue, bool) {
	for i := range *am.orig {
		akv := &(*am.orig)[i]
		if akv.Key == key {
			return AttributeValue{&akv.Value}, true
		}
	}
	return AttributeValue{nil}, false
}

```

I debated using nil on my first pass, but the resulting code wasn't as clean.  I don't mind refactoring it if we're worried about the performance hit.

Reply:
Found a way to avoid using nil at all -- ended up not as gross as my first pass. Still not a huge fan of the triple nested `if`, but I can live with it 😄

Reply:
Thanks - sorry yeah forgot it's not a native map